stratum/hal/lib/barefoot/bf.proto

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020-present Open Networking Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+option cc_generic_services = false;
+
+package stratum.hal.barefoot;
+
+import "p4/config/v1/p4info.proto";
+
+// BfDeviceConfig contains all the information of a P4 program that can be
+// loaded into the SDE.
+message BfDeviceConfig {
+  message P4Program {
+    string name = 1; // For display purposes only.
+    bytes bfrt = 2;  // JSON
+    p4.config.v1.P4Info p4info = 3;
+    repeated P4Pipeline pipelines = 4;
+  }
+
+  message P4Pipeline {
```

Comment:
Rename to P4Profile?

Reply:
The structure of the .conf file is:

```{
  "p4_devices": [{
    "p4_programs": [{
        "p4_pipelines": [{
            "context": "/tmp/pipe/context.json",
            "pipe_scope": [
                0,
                1,
                2,
                3
            ],
            "p4_pipeline_name": "pipe",
            "config": "/tmp/pipe/tofino.bin",
            "path": "/tmp"
        }],
        "bfrt-config": "/tmp/bfrt.json",
        "program-name": "my_prog"
    }],
    "device-id": 1
  }]
}
```

Two things:

1. It seems beneficial to mirror the conf JSON schema.
2. The goal of this proto is to encapsulate things at the `p4_programs` node.

Reply:
Good points, my initial thought was staying closer to the bfruntime.proto definition.

Reply:
That might be better... It would probably also mean that it is easier for us to convince the Intel team to support that format natively.

Should we use this? 

```
message ForwardingPipelineConfig {
  string p4_name = 1;               // P4 program name
  bytes bfruntime_info = 2;         // BF-RT info json file contents
  message Profile {                 // P4 Pipeline Profile
    string profile_name = 1;        // profile name
    bytes context = 2;              // context json file contents
    bytes binary = 3;               // Binary to execute
    repeated uint32 pipe_scope = 4; // Array of pipe_scope.
  }
  repeated Profile profiles = 3;
}
```

Reply:
Let's gather some experience and feedback, then we can make informed changes.